src/ledger/LedgerManagerImpl.h

```diff
@@ -413,6 +409,9 @@ class LedgerManagerImpl : public LedgerManager
                                     Config const& config);
 
     State mState;
+    bool mIsBooting{false};
+    std::condition_variable mBootingCV;
```

Comment:
let's avoid the complexity of getting concurrency right here: how about LM posting rebuild task to background, then background posting a back to main. The callback would set LM state back to either "synced" or "catching up". Then LedgerApplyManager can apply all buffered ledgers whenever a new buffered ledger comes in. This way no locks are needed.